openscap_report/report_generators/json.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2022, Red Hat, Inc.
+# SPDX-License-Identifier: LGPL-2.1-or-later
+import json
+import logging
+from io import BytesIO
+from pathlib import Path
+
+from jinja2 import Environment, FileSystemLoader
+
+from .report_generator import ReportGenerator
+
+
+class JSONReportGenerator(ReportGenerator):
+    def __init__(self, parser):
+        super().__init__(parser)
+        self.file_loader = FileSystemLoader(str(Path(__file__).parent / "json_templates"))
+        self.env = Environment(loader=self.file_loader)
+        self.env.trim_blocks = True
+        self.env.lstrip_blocks = True
+
+    def generate_report(self, debug_setting):
+        logging.warning("JSON Format is experimental output!")
+        template = self.env.get_template("base.json")
+        json_data = json.loads(template.render(report=self.report, debug_setting=debug_setting))
```

Comment:
I'm not quite sure that Jinja is needed here. You can just filter out keys from the report dict and then export it to JSON. Even if in the future there would be some more sophisticated logic (like keeping report.foo.bar and report.foo.baz but not report.foo.others) it'd be easier to just write a more complicated filter function rather than to siphon the whole report through Jinja engine.

Reply:
You're right.